README.md

```diff
@@ -116,6 +116,47 @@ pytest
 pytest --nbmake --nbmake-timeout=3000 examples/*ipynb
 ```
 
+To add new testing data, such as a new baseline for this block:
+
+```py
+dataset_name = "cthead1"
+image = input_images[dataset_name]
+baseline_name = "2_4/XARRAY_COARSEN"
+multiscale = to_multiscale(image, [2, 4], method=Methods.XARRAY_COARSEN)
+verify_against_baseline(dataset_name, baseline_name, multiscale)
+```
+
+Serialize the result:
+
+```py
+dataset_name = "cthead1"
+image = input_images[dataset_name]
+baseline_name = "2_4/XARRAY_COARSEN"
+multiscale = to_multiscale(image, [2, 4], method=Methods.XARRAY_COARSEN)
+
+store = DirectoryStore(
+    DATA_PATH / f"baseline/{dataset_name}/{baseline_name}", dimension_separator="/"
+)
+multiscale.to_zarr(store)
+
+verify_against_baseline(dataset_name, baseline_name, multiscale)
+```
+
+Run the tests to generate the output.
+
+Once the new test data is present locally, upload the result to IPFS:
+
+```sh
+npm install -g @web3-storage/w3
+# Get an upload token from https://web3.storage
+w3 token
+w3 put ./test/data --no-wrap --name multiscale-spatial-image-topic-name --hidden
```

Comment:
Is this uploading/replacing the entire test data archive every time? What happens if we accidentally overwrite a baseline file with a test output? Or if we intentionally replace a baseline?

Reply:
> Is this uploading/replacing the entire test data archive every time? 

It does upload the entire testing tree, but it does not replace.

Reply:
Not sure I understand. Are upstream, existing data files ignored? If so, why are they included in the upload?

Reply:
When the folder is uploaded, the folder is uploaded (identified by its contents) and all its contents recursively (a tree of folders and files). Each one of these folders and files has a verifiable hash derived from its contents. The content-addressed storage will not include duplicates (the duplicates have the same hash / content identifier (CID)). However, when you say "pin", i.e. keep, the root folder, the storage service holds on to the entire versioned tree.

Reply:
I think that makes sense to me. Thanks for the explanation.